packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -679,6 +734,107 @@ const Headers = ({ titles, classes, headerClasses }: { titles: string[], headerC
+      slug
+    }
+  }
+`, {
+  variables: { userId: currentUser?._id, targetUserId: targetUserId, limit : 4 },
+  skip: !currentUser
+});
+
+const { loading: commentsLoading, error: commentsError, data: commentsData } = useQuery(gql`
+  query UsersRecommendedCommentsOfTargetUser($userId: String!, $targetUserId: String!, $limit: Int!) {
+    UsersRecommendedCommentsOfTargetUser(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+      _id
+      postId
+      contents {
+        html
+        plaintextMainText
+      }
+    }
+  }
+`, {
+  variables: { userId: currentUser?._id, targetUserId: targetUserId, limit : 3 },
+  skip: !currentUser
+});
+
+const topTags:TagWithCommentCount[] | undefined = tagData?.UserTopTags;
+const readPosts:PostYouveRead[] | undefined = postsData?.UsersReadPostsOfTargetUser
+const recommendedComments:RecommendedComment[] | undefined = commentsData?.UsersRecommendedCommentsOfTargetUser
+
+if (!currentUser || !topTags || !readPosts || !recommendedComments) return <Loading />;
+const tagsSentence = topTags.slice(0, 4).map(tag => tag.tag.name).join(', ');
+const numRecommendations = (readPosts?.length ?? 0) + (recommendedComments?.length ?? 0) + (tagsSentence === "" ? 0 : 1);
+const numShown = isExpanded ? numRecommendations : 2
+const numHidden = Math.max(0, numRecommendations - numShown);
+
+const allRecommendations:{reactIconName:string, prefix:string, Content:JSX.Element}[] = [
+  ...(topTags.length > 0 ? [{reactIconName: "examples", prefix: "top tags: ", Content: <>{tagsSentence}</>}] : []),
+  ...readPosts.map(post => ({reactIconName: "elaborate", prefix: "post: ", Content: 
+    <PostsTooltip postId={post._id}>
+      <Link to={postGetPageUrl(post)}> {post.title} </Link>
+    </PostsTooltip>})),
+  ...recommendedComments.map(comment => ({reactIconName: "elaborate", prefix: "comment: ", Content: <CommentView comment={comment} />}))
+]
+
+const toggleExpansion = () => {
+  setIsExpanded(!isExpanded);
+  captureEvent("toggle_expansion_inside_match_form")
+};
+
+if (allRecommendations.length === 0) return null;
+
+return (
+  <AnalyticsContext pageElementContext={'userRecommendedContent'} >
+    <div className={classes.contentRecommendationsCard} >
+      <div className={classes.cardTitle}>
+        Recommended content
+      </div>
+      <div className={classNames(classes.recommendedContentContainer, {
+        [classes.recommendedContentContainerExpandedMobile]: isExpanded})}>
+        <div className={classes.contentRecommendationsList}>
+          {allRecommendations.slice(0, numShown).map( (item, index) => <TopicSuggestion key={index} reactIconName={item.reactIconName} prefix={item.prefix} Content={item.Content} isExpanded={isExpanded} />) } 
+        </div>
+        <div className={classes.recommendedContentRightContainer}>
+          {(allRecommendations && allRecommendations.length > 0) && <ExpandCollapseText isExpanded={isExpanded} numHidden={numHidden} toggleExpansion={toggleExpansion} />}
```

Comment:
`(allRecommendations && allRecommendations.length > 0)` feels unnecessary given the check at L817

Reply:
legit